src/data.rs

```diff
@@ -60,6 +67,24 @@ impl Data {
             Data::PrivAppendable(ref data) => data.identifier(),
         }
     }
+
+    /// Validate data size.
+    pub fn validate_size(&self) -> bool {
+        let serialised_data = match serialisation::serialise(&self) {
```

Comment:
we store serialized DATA into chunkstore, so I think the limit shall be applied to whole instead of payload only.
regarding the duplicated serialization, yes, that is an issue and I will put a TODO comment in.